addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -178,38 +294,116 @@ class ComposerTextInput extends Component {
      * @param {KeyboardEvent} ev
      */
     _onKeydownTextareaEnter(ev) {
-        if (!this.props.hasSendOnEnterEnabled) {
-            return;
+        if (this.composer.suggestedPartners.length > 0) {
+            ev.preventDefault();
+        } else {
+            if (!this.props.hasSendOnEnterEnabled) {
+                return;
+            }
+            if (ev.shiftKey) {
+                return;
+            }
+            if (this.env.messaging.device.isMobile) {
+                return;
+            }
+            this.trigger('o-keydown-enter');
+            ev.preventDefault();
         }
-        if (ev.shiftKey) {
-            return;
-        }
-        if (this.env.messaging.device.isMobile) {
-            return;
+    }
+
+    /**
+     * Key events management is performed in a Keyup to avoid intempestive RPC calls
+     *
+     * @private
+     * @param ev
+     */
+    _onKeyupTextarea(ev) {
+        switch (ev.key) {
+            // ESCAPE: close mention suggestions
+            case 'Escape':
+                this._onKeyupTextareaEscape(ev);
+                break;
+            // ENTER, HOME, END, UP, DOWN, PAGE UP, PAGE DOWN, TAB: check if navigation in mention suggestions
+            case 'Enter':
+                if (this.composer.suggestedPartners.length > 0) {
```

Comment:
Considering the repeated use of the long text `this.composer.suggestedPartners.length > 0` in this method, I would create a const for it at the top (`hasSuggestedPartners`, or similar name), this could even be a getter on the component.

Reply:
I like the idea of a getter but shouldn't it be in the entity instead of the component ?

Reply:
But I like the idea that it should be in the entity, this is conceptually the right place, but then it should be a computed field and not a getter.

If the helper is on the component, the rest of the component code can just do `this.hasSuggestedPartners`, if the helper is on the entity, the component has to do `this.composer.hasSuggestedPartners` which is not really shorter than the non-getter version...

Start by doing the one in the entity, and if it still feels too long here, we can add one in the component that is using the result of the one in entity.